### PR TITLE
Editor can publish articles

### PR DESCRIPTION
**Pull request by NiciP**
#### Description
As an editor,
In order to make sure that the articles have good standards
I would like only editors to be able to publish articles.

Changes proposed in this pull request:
* Create namespace for Editor
* Add functionality for publishing articles 
* Add relationship between editor and article
#### What I have learned working on this feature: 
Changing the status of a column in a db table by interaction by user

### CODE REVIEW

File: features/visitor_can_see_articles_on_index_page.feature [hunk 1]
```diff
@@ -7,13 +7,14 @@ Feature: List articles on the landing page
...
+            | per@mail.se    | password | journalist |
 
         And the following articles exist
-            | title                                     | lead                        | content                                                     | category     | user           |
-            | Voted best mead recipe                    | Restaurant wins prize       | Restaurant wins prize for best mead in Sweden               | Lifestyle    | jocke@craft.se |
-            | Ancient viking grave discovered           | Kids came across sword      | Kids come across sword protruding from the earth            | Breaking News| jocke@craft.se |
-            | Drinking wine improves general health     | Drink wine today!           | Studies show that wine is good for your heart               | Health       | jocke@craft.se |
+            | title                                 | lead                   | content                                          | category      | user           | approved |
+            | Voted best mead recipe                | Restaurant wins prize  | Restaurant wins prize for best mead in Sweden    | Lifestyle     | jocke@craft.se | true     |
```
Comment (vick3d) [inline features/visitor_can_see_articles_on_index_page.feature]: delete approved column, factory bot is handling it for value true

**Review by vick3d — CHANGES_REQUESTED** [review]

File: features/visitor_can_see_articles_on_index_page.feature (hunk 1 above)
Comment (NiciP) [reply features/visitor_can_see_articles_on_index_page.feature]: yes thats right, done

**Review by yanqiuzhang — APPROVED** [review]

**Review by vick3d — APPROVED** [review]

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -6,9 +6,9 @@ class ArticlesController < ApplicationController
     def index
         @categories = Category.all
         if params[:category].present?
-            @articles = Category.find_by(name: params[:category]).articles
+            @articles = Category.find_by(name: params[:category]).articles.where(approved: true)
```
Comment (magnus-thor) [inline app/controllers/articles_controller.rb]: `.articles.where(approved: true)`could you not have used `.articles.approved` like you do in the else step?

File: app/controllers/journalist/articles_controller.rb [hunk 3]
```diff
@@ -21,6 +21,7 @@ def create
     end
 
     def edit
+        @article = Article.find(params[:id])
```
Comment (magnus-thor) [inline app/controllers/journalist/articles_controller.rb]: any reason you added this in?

File: app/controllers/journalist/articles_controller.rb [hunk 4]
```diff
@@ -32,6 +33,8 @@ def update
     end
 
     def destroy
+        @article = Article.find(params[:id])
```
Comment (magnus-thor) [inline app/controllers/journalist/articles_controller.rb]: again, any reason you added this in?

File: app/views/editor/articles/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+<p>
+    <strong>Text:</strong>
+    <%= @article.content %>
+</p>
+
+<%= link_to "Back", editor_articles_path %> 
+
+<% if @article.approved == false %>
+<%= link_to "Publish", editor_article_path(@article, approved: true), method: :patch  %>
```
Comment (magnus-thor) [inline app/views/editor/articles/show.html.erb]: watch the code formatting.

File: db/seeds.rb [hunk 6]
```diff
@@ -8,18 +8,19 @@
...
 category_3 = Category.create(name: "Politics")
 category_4 = Category.create(name: "Health")
 category_5 = Category.create(name: "Sport")
 
-article_1 = Article.create(title: "Voted best mead recipe", lead: "Restaurant wins prize", content: "Restaurant wins prize for best mead in Sweden")
-article_2 = Article.create(title: "Ancient viking grave discovered", lead: "Kids came across sword", content: "Kids come across sword protruding from the earth")
-article_3 = Article.create(title: "New Prime Minister", lead: "Today is a new day", content: "The country is on a better path")
-article_4 = Article.create(title: "Drinking wine improves general health", lead: "Drink wine today!", content: "Studies show that wine is good for your heart")
-article_5 = Article.create(title: "Hammarby! Hammarby!", lead: "Best team ever!", content: "Home is where the heart is")
+article_1 = Article.create(title: "Voted best mead recipe", lead: "Restaurant wins prize", content: "Restaurant wins prize for best mead in Sweden", user_id: 1)
```
Comment (magnus-thor) [inline db/seeds.rb]: Dont hard code the user id.

**Review by magnus-thor — CHANGES_REQUESTED** [review]

File: app/controllers/articles_controller.rb (hunk 2 above)
Comment (NiciP) [reply app/controllers/articles_controller.rb]: yes thatll work, changed it

File: app/controllers/journalist/articles_controller.rb (hunk 3 above)
Comment (NiciP) [reply app/controllers/journalist/articles_controller.rb]: no we didnt use it in the end

File: app/controllers/journalist/articles_controller.rb (hunk 4 above)
Comment (NiciP) [reply app/controllers/journalist/articles_controller.rb]: also removed it

File: app/views/editor/articles/show.html.erb (hunk 5 above)
Comment (NiciP) [reply app/views/editor/articles/show.html.erb]: ok, cleaned it up

File: db/seeds.rb (hunk 6 above)
Comment (NiciP) [reply db/seeds.rb]: ok

**Review by magnus-thor — APPROVED** [review]

**Pull request merged**